services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/VxlanFlowSpec.groovy

```diff
@@ -0,0 +1,418 @@
+        exc.rawStatusCode == 404
+        // TODO(andriidovhan) fix errorMessage when the 2587 issue is fixed
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: Not enough bandwidth found or path not found. Failed to find path with " +
+                "requested bandwidth=$flow.maximumBandwidth: " +
+                "Switch $switchPair.dst.dpId doesn't have links with enough bandwidth"
+    }
+
+    @Tags(HARDWARE)
+    def "System allows to create/update encapsulation type for an one-switch flow\
+(#encapsulationCreate.toString() -> #encapsulationUpdate.toString())"() {
+        when: "Try to create an one-switch flow"
+        def noviflowSwitchIds = topology.activeSwitches.findAll { it.noviflow }*.dpId
+        def sw = topology.activeTraffGens*.switchConnected.find {
+            noviflowSwitchIds.contains( it.dpId)
+        } ?: assumeTrue("Should be at least one active traffgen connected to NoviFlow switch",false)
+        def flow = flowHelper.singleSwitchFlow(sw)
+        flow.encapsulationType = encapsulationCreate
+        northbound.addFlow(flow)
+
+        then: "Flow is created with the #encapsulationCreate.toString() encapsulation type"
+        def flowInfo1 = northbound.getFlow(flow.id)
+        flowInfo1.encapsulationType == encapsulationCreate.toString().toLowerCase()
+
+        //TODO(andriidovhan) check rules (tunnel-id) when pr2503 is merged
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.id).each { direction -> assert direction.asExpected }
+
+        and: "Flow is pingable"
+        def responsePing1 = northbound.pingFlow(flow.id, new PingInput())
+        responsePing1.forward.pingSuccess
+        responsePing1.reverse.pingSuccess
+
+        when: "Try to update the encapsulation type to #encapsulationUpdate.toString()"
+        northbound.updateFlow(flow.id, flow.tap { it.encapsulationType = encapsulationUpdate })
+
+        then: "The encapsulation type is changed to #encapsulationUpdate.toString()"
+        def flowInfo2 = northbound.getFlow(flow.id)
+        flowInfo2.encapsulationType == encapsulationUpdate.toString().toLowerCase()
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.id).each { direction -> assert direction.asExpected }
+
+        and: "Flow is pingable"
+        def responsePing2 = northbound.pingFlow(flow.id, new PingInput())
+        responsePing2.forward.pingSuccess
+        responsePing2.reverse.pingSuccess
+
+        //TODO(andriidovhan) check rules(tunnel-id) when pr2503 is merged
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        where:
+        encapsulationCreate | encapsulationUpdate
+        FlowEncapsulationType.TRANSIT_VLAN | FlowEncapsulationType.VXLAN
+        FlowEncapsulationType.VXLAN | FlowEncapsulationType.TRANSIT_VLAN
+    }
+
+    @Tags(VIRTUAL)
+    def "System doesn't allow to create a flow with 'VXLAN' encapsulation on an OVS switch"() {
```

Comment:
Remove 'virtual', refactor test to be `System doesn't allow to create a flow with 'VXLAN' encapsulation on a non-noviflow switch`
Same for other 'ovs' tests

Reply:
fixed